api_v3/lib/types/EntryServerNode/base/KalturaEntryServerNodeArray.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * @package api
+ * @subpackage objects
+ */
+class KalturaEntryServerNodeArray extends KalturaTypedArray
+{
+	public static function fromDbArray($arr)
+	{
+		$newArr = new KalturaEntryServerNodeArray();
+		foreach($arr as $obj)
+		{
+			/* @var $obj EntryServerNode */
+			$nObj = EntryServerNodePeer::getInstanceByType($obj);
```

Comment:
implement getInstanceByType on KalturaEntryServerNode, implement IApiObjectFactory and call the method accordingly.